src/core/Remixer.ts

```diff
@@ -305,16 +305,19 @@ class Remixer {
    * @static
    * @param {Variable} variable The variable to clone and update.
    * @param {any} selectedValue The new selected value.
+   * @return {Variable} The cloned variable with updated selected value.
    */
-  static cloneAndUpdateVariable(variable: Variable, selectedValue: any): void {
+  static cloneAndUpdateVariable(variable: Variable, selectedValue: any): Variable {
     // First make sure selected value is in proper format.
     selectedValue = variable.formatValue(selectedValue);
 
     if (variable.selectedValue !== selectedValue) {
```

Comment:
nit: it would be easier to read if you did the opposite, return right away if selectedValues are the same